projects/packages/forms/src/dashboard/inbox/dataviews/index.js

```diff
@@ -0,0 +1,289 @@
+						data?.find( record => getItemId( record ) === items[ items.length - 1 ] )
+				);
+			}
+			setSearchParams( previouSearchParams => {
+				const _serachParams = new URLSearchParams( previouSearchParams );
+				if ( items.length ) {
+					_serachParams.set( 'r', items.join( ',' ) );
+				} else {
+					_serachParams.delete( 'r' );
+				}
+				return _serachParams;
+			} );
+		},
+		[ data, setSearchParams, isMobile ]
+	);
+	// Because selection is in sync with the URL and data takes some time to load,
+	// We need to carefully (avoid infinite loops by always updating the state)
+	// set the sidePanelItem when we have data and selection.
+	// We don't need to do this in `mobile`,  because we don't render the side panel.
+	if ( ! isMobile && !! data && !! selection.length ) {
+		const firstValidSelection = selection.find( id =>
+			data.some( record => getItemId( record ) === id )
+		);
+		const recordToShow = data?.find( record => getItemId( record ) === firstValidSelection );
+		if ( ! sidePanelItem && recordToShow ) {
+			setSidePanelItem( recordToShow );
+		} else if ( !! sidePanelItem && ! recordToShow ) {
+			// This case handles the case where we were having a side panel item
+			// visible but the data have changed and the item is not there anymore.
+			setSidePanelItem();
+		}
+	}
+	const paginationInfo = useMemo(
+		() => ( { totalItems, totalPages } ),
+		[ totalItems, totalPages ]
+	);
+	const fields = useMemo(
+		() => [
+			{
+				id: 'from',
+				label: __( 'From', 'jetpack-forms' ),
+				getValue: ( { item } ) => {
+					return (
+						decodeEntities( item.author_name ) || item.author_email || item.author_url || item.ip
+					);
+				},
+			},
+			{
+				id: 'date',
+				label: __( 'Date', 'jetpack-forms' ),
+				render: ( { item } ) => dateI18n( 'M j, Y', item.date ),
+				elements: ( filterOptions?.date || [] ).map( _filter => {
+					const date = new Date();
+					date.setDate( 1 );
+					date.setMonth( _filter.month - 1 );
+					return {
+						label: `${ dateI18n( 'F', date ) } ${ _filter.year }`,
+						value: `${ _filter.year }/${ _filter.month }`,
+					};
+				} ),
+				filterBy: { operators: [ 'is' ] },
+				enableSorting: false,
+			},
+			{
+				id: 'source',
+				label: __( 'Source', 'jetpack-forms' ),
+				render: ( { item } ) => {
+					return (
+						<ExternalLink href={ item.entry_permalink }>
+							{ decodeEntities( item.entry_title ) || getPath( item ) }
+						</ExternalLink>
+					);
+				},
+				elements: ( filterOptions?.source || [] ).map( source => ( {
+					value: source.id,
+					label: source.title,
+				} ) ),
+				filterBy: { operators: [ 'is' ] },
+				enableSorting: false,
+			},
+			{ id: 'ip', label: __( 'IP Address', 'jetpack-forms' ), enableSorting: false },
+		],
+		[ filterOptions ]
+	);
+	const actions = useMemo( () => {
+		return [
+			viewAction,
+			markAsSpamAction,
+			markAsNotSpamAction,
+			moveToTrashAction,
+			restoreAction,
+			deleteAction,
+		];
+	}, [] );
```

Comment:
Nit: can we define this array outside of the component render function, since it looks like a constant value?